### ISSUE TITLE
AssertLinesMatch NullPointerException during FastForward evaluation

### ISSUE THREAD

**Issue by jimmorgan** [junit-team/junit5#1640]
## Overview

AssertLinesMatch#matches has a bug that will cause a NullPointerException to be thrown after evaluating a FastForward match if there are more expected lines after the FastForward match then remain in the actual results.  This only manifests itself if the expected list size is equal to or larger than the actual list size.  The reason this fails is because after matching the fast forward region, the actual list is now empty but AssertLinesMatch doesn't check for that condition and sends the (null) value from the actualDeque.peek call to AssertLineMatch#matches.

The solution is to simply add a check for `null` in matches.  The return statement could be modified to:
```java
return (actualLine != null) && actualLine.matches(expectedLine);
```
An example that illustrates this error can be seen in the following code:
```java
List<String> expected = Arrays.asList("first line", ">> fails, because final line is missing >>", "last line", "not present");
List<String> actual = Arrays.asList("first line", "first skipped", "second skipped", "last line");
Assertions.assertLinesMatch(expected, actual); // <- throws NullPointerException
```

## Deliverables

A pull request with this issue number as the branch name will be submitted that contains the change to AssertLineMatch#matches and a test method added to AssertLineMatchAssertionsTest to illustrate the bug is fixed.

**Comment by marcphilipp** [comment]
Thanks for the bug report! Would you like to submit a PR with your fix?

**Comment by sbrannen** [comment]
> A pull request with this issue number as the branch name will be submitted that contains the change to AssertLineMatch#matches and a test method added to AssertLineMatchAssertionsTest to illustrate the bug is fixed.

@jimmorgan, yes, please submit a PR, and we'll merge it in as soon as possible.

Please remember to document the fix in the 5.4 M1 Release Notes.

Thanks in advance!

**Comment by jimmorgan** [comment]
Submitted at PR 1642.

**Comment by sbrannen** [comment]
_in progress_ by @jimmorgan in  #1642

**Comment by sbrannen** [comment]
Fixed in `master` in 07d52896cabb88362951de77070d35c81f5512d8.
